From bbd0e91f6f08fea7a0bf1711a96140a8ef0be51a Mon Sep 17 00:00:00 2001
From: cnjn <1779895255@qq.com>
Date: Sat, 22 Aug 2026 20:00:22 +0800
Subject: [PATCH 15/16] arm64: dts: zte: enable SR1010 networking

Consolidate 1 validated ZX279133 development commits into one functional stage.

Original range: 635b92bfa1e92d2e1497c44ab3b5e21467caa6b6..635b92bfa1e92d2e1497c44ab3b5e21467caa6b6

Signed-off-by: cnjn <1779895255@qq.com>
---
 arch/arm64/boot/dts/zte/zx279133-sr1010.dts | 136 ++++++++++++++++++++
 arch/arm64/boot/dts/zte/zx279133.dtsi       |  73 ++++++++++-
 2 files changed, 203 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/zte/zx279133-sr1010.dts b/arch/arm64/boot/dts/zte/zx279133-sr1010.dts
index b25bb50ae633be72652f19002a918322ab051c85..4cd2d18bd49973ac1ed4c33405d688617321efa9 100644
--- a/arch/arm64/boot/dts/zte/zx279133-sr1010.dts
+++ b/arch/arm64/boot/dts/zte/zx279133-sr1010.dts
@@ -26,6 +26,27 @@ memory@80000000 {
 		reg = <0x0 0x80000000 0x0 0x20000000>;
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		se_hash_reserved: buffer@9e100000 {
+			reg = <0x0 0x9e100000 0x0 0x1000000>;
+			no-map;
+		};
+
+		idm_reserved: buffer@9d700000 {
+			reg = <0x0 0x9d700000 0x0 0xa00000>;
+			no-map;
+		};
+
+		bmu_reserved: buffer@9f100000 {
+			reg = <0x0 0x9f100000 0x0 0x0f00000>;
+			no-map;
+		};
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 		label = "SR1010 reset button";
@@ -79,6 +100,121 @@ &pvt {
 	status = "okay";
 };
 
+&pon_serdes {
+	status = "okay";
+};
+
+&uni_serdes {
+	status = "okay";
+};
+
+&nppt {
+	memory-region = <&bmu_reserved>, <&se_hash_reserved>, <&idm_reserved>;
+	memory-region-names = "bmu", "se-hash", "idm";
+	phy-handle = <&zx279051>;
+	status = "okay";
+
+	lan_cpu: ethernet-lan-cpu {
+		compatible = "zte,zx279133-lan-conduit";
+	};
+
+	switch {
+		compatible = "zte,zx279133-rtl8372n";
+		zte,mdio-handle = <&rtl8372n_smi>;
+		reset-gpios = <&gpio1 1 GPIO_ACTIVE_LOW>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@4 {
+				reg = <4>;
+				label = "lan4";
+				phy-mode = "internal";
+
+				fixed-link {
+					speed = <2500>;
+					full-duplex;
+				};
+			};
+
+			port@5 {
+				reg = <5>;
+				label = "lan3";
+				phy-mode = "internal";
+
+				fixed-link {
+					speed = <2500>;
+					full-duplex;
+				};
+			};
+
+			port@6 {
+				reg = <6>;
+				label = "lan2";
+				phy-mode = "internal";
+
+				fixed-link {
+					speed = <2500>;
+					full-duplex;
+				};
+			};
+
+			port@7 {
+				reg = <7>;
+				label = "lan1";
+				phy-mode = "internal";
+
+				fixed-link {
+					speed = <2500>;
+					full-duplex;
+				};
+			};
+
+			port@8 {
+				reg = <8>;
+				ethernet = <&lan_cpu>;
+				phy-mode = "10gbase-r";
+
+				fixed-link {
+					speed = <10000>;
+					full-duplex;
+				};
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	rtl8372n_smi: switch-controller@1d {
+		compatible = "realtek,rtl8372n";
+		reg = <0x1d>;
+	};
+};
+
+&mdio1 {
+	status = "okay";
+
+	zx279051: ethernet-phy@1 {
+		compatible = "ethernet-phy-id0000.84b9";
+		reg = <1>;
+		max-speed = <2500>;
+		reset-gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+		reset-assert-us = <100000>;
+		reset-deassert-us = <100000>;
+	};
+};
+
+&xpcs0 {
+	status = "okay";
+};
+
+&xpcs1 {
+	status = "okay";
+};
+
 &watchdog1 {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/zte/zx279133.dtsi b/arch/arm64/boot/dts/zte/zx279133.dtsi
index 7d7c9cdbf43d0b790aafec66c751dcb466ef9540..ca6299bf2b263719dd83fe7603310ce70eb37704 100644
--- a/arch/arm64/boot/dts/zte/zx279133.dtsi
+++ b/arch/arm64/boot/dts/zte/zx279133.dtsi
@@ -4,6 +4,7 @@
 #include <dt-bindings/clock/zte,zx279133-topcrm.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/reset/zte,zx279133-lsp.h>
+#include <dt-bindings/reset/zte,zx279133-topcrm.h>
 
 / {
 	compatible = "zte,zx279133";
@@ -112,6 +113,11 @@ topcrm: clock-controller@10e10000 {
 			#clock-cells = <1>;
 		};
 
+		idm_cci: syscon@10e00078 {
+			compatible = "zte,zx279133-idm-cci", "syscon";
+			reg = <0x0 0x10e00078 0x0 0x8>;
+		};
+
 		top_reset: reset-controller@10e10060 {
 			compatible = "zte,zx296718-reset";
 			reg = <0x0 0x10e10060 0x0 0x4>;
@@ -418,12 +424,67 @@ watchdog1: watchdog@14f0a000 {
 			status = "disabled";
 		};
 
-		nppt: packet-processor@19000000 {
+		pon_serdes: phy@16000000 {
+			compatible = "zte,zx279133-pon-serdes";
+			reg = <0x0 0x16000000 0x0 0x10000>,
+			      <0x0 0x16010000 0x0 0x10000>,
+			      <0x0 0x10e100c0 0x0 0x8>;
+			reg-names = "serdes", "pll", "topcrm-mode";
+			clocks = <&topcrm ZX279133_TOPCRM_CLK_PON_SERDES_PCLK>;
+			clock-names = "pclk";
+			resets = <&local_reset ZX279133_LOCAL_RESET_PON_SERDES>,
+				 <&local_reset ZX279133_LOCAL_RESET_PON_SERDES_APB>;
+			reset-names = "serdes", "apb";
+			zte,topcrm = <&topcrm>;
+			#phy-cells = <0>;
+			status = "disabled";
+		};
+
+		pon_route: syscon@17000080 {
+			compatible = "zte,zx279133-pon-route", "syscon";
+			reg = <0x0 0x17000080 0x0 0x4>;
+		};
+
+		uni_serdes: phy@16100000 {
+			compatible = "zte,zx279133-uni-serdes";
+			reg = <0x0 0x16100000 0x0 0x100000>;
+			clocks = <&topcrm ZX279133_TOPCRM_CLK_UNI_SERDES_PCLK>,
+					 <&topcrm ZX279133_TOPCRM_CLK_UNI_SERDES_50M>;
+			clock-names = "pclk", "ref50m";
+			resets = <&local_reset ZX279133_LOCAL_RESET_UNI_SERDES_RX>,
+				 <&local_reset ZX279133_LOCAL_RESET_UNI_SERDES_TX>;
+			reset-names = "rx", "tx";
+			#phy-cells = <0>;
+			status = "disabled";
+		};
+
+		nppt: ethernet@19000000 {
 			compatible = "zte,zx279133-nppt";
-			reg = <0x0 0x19000000 0x0 0x1000000>;
-			interrupts = <GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&topcrm ZX279133_TOPCRM_CLK_PON_WOE1_WCLK>;
-			clock-names = "wclk";
+			reg = <0x0 0x19000000 0x0 0x1000000>,
+			      <0x0 0x18000000 0x0 0x1000000>;
+			reg-names = "nppt", "pps";
+			interrupts = <GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "nppt", "rx", "idm", "buffer-release",
+					  "local-test";
+			clocks = <&topcrm ZX279133_TOPCRM_CLK_PON_IDM_ACLK>,
+				 <&topcrm ZX279133_TOPCRM_CLK_PON_TM_ACLK>,
+				 <&topcrm ZX279133_TOPCRM_CLK_PON_PCLK>,
+				 <&topcrm ZX279133_TOPCRM_CLK_PON_SMAC_WCLK>,
+				 <&topcrm ZX279133_TOPCRM_CLK_PON_WOE1_WCLK>,
+				 <&topcrm ZX279133_TOPCRM_CLK_PON_MAC_WCLK>;
+			clock-names = "idm-aclk", "tm-aclk", "pclk", "smac-wclk",
+				      "woe1-wclk", "mac-wclk";
+			pcs-handle = <&xpcs1>;
+			zte,lan-pcs-handle = <&xpcs0>;
+			phys = <&pon_serdes>, <&uni_serdes>;
+			phy-names = "serdes", "lan-serdes";
+			zte,pon-route = <&pon_route>;
+			zte,idm-cci = <&idm_cci>;
+			phy-mode = "2500base-x";
 			status = "disabled";
 		};
 
@@ -442,7 +503,7 @@ xpcs1: ethernet-pcs@1b000000 {
 			reg = <0x0 0x1b000000 0x0 0x800000>;
 			reg-names = "direct";
 			reg-io-width = <4>;
-			clocks = <&topcrm ZX279133_TOPCRM_CLK_UNI_SERDES_PCLK>;
+			clocks = <&topcrm ZX279133_TOPCRM_CLK_PON_SERDES_PCLK>;
 			clock-names = "csr";
 			status = "disabled";
 		};
